modules/client/src/main/java/org/apache/ignite/internal/jdbc/JdbcResultSet.java

```diff
@@ -0,0 +1,1893 @@
+
+            rowsIter = rows != null ? rows.iterator() : null;
+        }
+        else
+            this.updCnt = updCnt;
+    }
+
+    /** {@inheritDoc} */
+    @Override public boolean next() throws SQLException {
+        ensureNotClosed();
+
+        if ((rowsIter == null || !rowsIter.hasNext()) && !finished) {
+            JdbcQueryFetchResult res = qryHandler.fetch(new JdbcQueryFetchRequest(cursorId, fetchSize));
+
+            if (!res.hasResults())
+                throw IgniteQueryErrorCode.createJdbcSqlException(res.err(), res.status());
+
+            rows = res.items();
+            finished = res.last();
+
+            rowsIter = rows.iterator();
+        }
+
+        if (rowsIter != null) {
+            if (rowsIter.hasNext()) {
+                curRow = rowsIter.next();
+
+                curPos++;
+
+                return true;
+            }
+            else {
+                rowsIter = null;
+                curRow = null;
+
+                return false;
+            }
+        }
+        else
+            return false;
+    }
+
+    /** {@inheritDoc} */
+    @Override public void close() throws SQLException {
+        close0();
+
+        if (closeStmt)
+            stmt.closeIfAllResultsClosed();
+    }
+
+    /**
+     * Close result set.
+     *
+     * @throws SQLException On error.
+     */
+    void close0() throws SQLException {
+        if (isClosed())
+            return;
+
+        try {
+            if (stmt != null && (!finished || (isQuery && !autoClose))) {
+                JdbcQueryCloseResult res = qryHandler.close(new JdbcQueryCloseRequest(cursorId));
+
+                if (!res.hasResults())
+                    throw IgniteQueryErrorCode.createJdbcSqlException(res.err(), res.status());
+            }
+        }
+        finally {
+            closed = true;
+        }
+    }
+
+    /** {@inheritDoc} */
+    @Override public boolean wasNull() throws SQLException {
+        ensureNotClosed();
+        ensureHasCurrentRow();
+
+        return wasNull;
+    }
+
+    /** {@inheritDoc} */
+    @Override public String getString(int colIdx) throws SQLException {
+        Object val = getValue(colIdx);
+
+        return val == null ? null : String.valueOf(val);
+    }
+
+    /** {@inheritDoc} */
+    @Override public boolean getBoolean(int colIdx) throws SQLException {
+        Object val = getValue(colIdx);
+
+        if (val == null)
+            return false;
+
+        Class<?> cls = val.getClass();
+
+        if (cls == Boolean.class)
+            return ((Boolean)val);
+        else if (val instanceof Number)
+            return ((Number)val).intValue() != 0;
+        else if (cls == String.class || cls == Character.class) {
+            try {
+                return Integer.parseInt(val.toString()) != 0;
+            }
+            catch (NumberFormatException e) {
+                throw new SQLException("Cannot convert to boolean: " + val, SqlStateCode.CONVERSION_FAILED, e);
```

Comment:
SQLDataException ?

Reply:
As I see in the javadoc of the exception, SQLDataException should be thrown when the SQLState class value is '22', or under vendor-specified conditions.  Here and other places we have 0700B error code - "data type transform function violation". 
I've looked at other database drivers (postgre, mysql), and none throw this type of error. But the documentation allows it, so I can replace, if you see it as the right thing to do.